### PR TITLE
Add timeout to process execution

### PR DESCRIPTION
**Pull request by steelbrain**
Fixes https://github.com/AtomLinter/atom-linter/issues/81

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 ## Upcoming
 
+*   Add `timeout` option to exec
```
Comment (Arcanemagus) [inline CHANGELOG.md]: Add ", default of 10 seconds." or something along those lines.

File: src/helpers.js [hunk 2]
```diff
@@ -91,8 +91,10 @@ export async function exec(
   const options: ExecOptions = assign({
     env: await consistentEnv.async(),
     stream: 'stdout',
-    throwOnStdErr: true
+    throwOnStdErr: true,
+    timeout: 10000 // ten seconds
```
Comment (Arcanemagus) [inline src/helpers.js]: "ten" -> "Ten"

Comment (Arcanemagus) [inline src/helpers.js]: Or probably better: Note that this is `ms` here.

File: src/helpers.js [hunk 3]
```diff
@@ -146,10 +149,16 @@ export async function exec(
...
-      }
+      } catch (_) { /* No Op */ }
     }
+    try {
+      spawnedProcess.process.stdin.end()
+    } catch (_) { /* No Op */ }
+    timeout = setTimeout(function () {
+      try {
+        spawnedProcess.kill()
+      } catch (_) { /* No Op */ }
```
Comment (Arcanemagus) [inline src/helpers.js]: Should this be a No Op?

Comment (Arcanemagus) [inline src/helpers.js]: Ah, probably not since it's likely the process has already ended at this point.

**Comment by Arcanemagus** [comment]
Any way to stop the timeout that will attempt to kill the process if it exits before then?

**Comment by steelbrain** [comment]
Now if someone supplies `Infinity` as timeout, we won't put any kind-of timeout on it

**Comment by Arcanemagus** [comment]
LGTM

**Pull request merged**